### PR TITLE
Add new Style/ZeroLengthPredicate cop

### PR DESCRIPTION
**Pull request by Drenmi**
This is my first go at implementing a cop, so please be gentle. :grimacing: 

I had a few things that I was unsure about while going about it, so I'll go over them here:

#### Naming

Some cops seem to be named after their offense, e.g. `Style/TrailingBlankLines`, while others seem named after the "better" practice, e.g. `Style/SymbolProc`.

In the end I settled on `Style/ZeroLengthPredicate`, as I couldn't come up with a name including `Empty` that didn't sound very generic.

#### Testing ambition

I wasn't quite sure how many tests to add, so I ended up adding tests for most positive case permutations. Should I also include the negative cases? In this cop, the `0` is hard coded, so I excluded such cases for now.

Thankful for any feedback.

:kissing_heart:

### CODE REVIEW

**Comment by Drenmi** [comment]
Not sure why `$ rvm use jruby-9.0.1.0 --install --binary --fuzzy` decided to randomly fail in my build ... :expressionless:

**Comment by alexdowad** [comment]
> Not sure why $ rvm use jruby-9.0.1.0 --install --binary --fuzzy decided to randomly fail in my build ...

It's not the first time. It doesn't need to hold this up from being merged either.

The code looks good, nothing jumps out at me that needs to be fixed up.

File: lib/rubocop/cop/style/zero_length_predicate.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      #   [1, 2, 3].empty?
+      #   "foobar".empty?
+      #   !hash.empty?
+      class ZeroLengthPredicate < Cop
+        ZERO_MSG = 'Use `empty?` instead of `%s == 0`.'.freeze
+        NONZERO_MSG = 'Use `!empty?` instead of `%s %s %s`.'.freeze
+
+        def on_send(node)
+          zero_length_predicate = zero_length_predicate(node)
+          if zero_length_predicate
```
Comment (bbatsov) [inline lib/rubocop/cop/style/zero_length_predicate.rb]: I'd add a few blank lines around those conditionals to increase the readability a bit.

**Comment by bbatsov** [comment]
Great first contribution! Apart from my tiny remark - you'll have to rebase on top of the current `master`.

**Comment by Drenmi** [comment]
@bbatsov: Thanks! I added space, rebased with `master`, and JRuby decided not to error out this time. :-)

**Comment by bbatsov** [comment]
:+1:

**Pull request merged**